File: src/main/webapp/src/app/customers/customer.ts
import { DateModel } from 'ng2-datepicker';


export class Customer {

  id: number;
  name: string;
  surname: string;
  email: string;
  address: string;
  city: string;
  personalNumber: string;
  phone: string;
  companyName: string;
  vatNumber: string;
  isActive = false;
  isCompany = false;
  createdAt: Date = null;
  bornDatePicker: DateModel = null;
  bornDate: Date = null;
  drivingLicenseDatePicker: DateModel = null;
  drivingLicenseDate: Date = null;
  firstPoliceDatePicker: DateModel = null;
  firstPoliceDate: Date = null;

  constructor(data) {

    Object.assign(this, data);

    if (data) {
      this.createdAt = data.createdAt ? new Date(data.createdAt) : null;
    }

  }

  getFullName(): string {

    if (this.name && this.surname) {
      return `${this.name} ${this.surname}`;
    }

    return '-';
  }

  getStatus() {
    return this.isActive ? 'Aktywny' : 'Nieaktywny';
  }

  prepareToSave() {
    if (this.bornDatePicker) {
      this.bornDate = this.bornDatePicker.momentObj.toDate();
    }

    if (this.drivingLicenseDatePicker) {
      this.drivingLicenseDate = this.drivingLicenseDatePicker.momentObj.toDate();
    }

    if (this.firstPoliceDatePicker) {
      this.firstPoliceDate = this.firstPoliceDatePicker.momentObj.toDate();
    }
  }
}

File: src/main/java/agency/entity/Customer.java
package agency.entity;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.validator.constraints.Email;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.config.EnableMongoAuditing;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import javax.validation.constraints.*;
import java.util.Date;

@Document(collection = "customers")
@EnableMongoAuditing
public class Customer {

    @Id @Getter
    private String id;

    @NotNull @Setter @Getter
    private String name;

    @NotNull @Setter @Getter
    private String surname;

    @Setter @Getter
    private String address;

    @Setter @Getter
    private String city;

    @Setter @Getter
    private String personalNumber;

    @Setter @Getter
    private Boolean isCompany;

    @Setter @Getter
    private String companyName;

    @Setter @Getter
    private String vatNumber;

    @Setter @Getter
    @NotNull @Email @Indexed(unique = true)
    private String email;

    @Setter @Getter
    private Boolean isActive;

    @Past @Setter @Getter
    private Date bornDate;

    @Setter @Getter
    private String phone;

    @Past @Setter @Getter
    private Date drivingLicenseDate;

    @Past @Setter @Getter
    private Date firstPoliceDate;

    @Setter @Getter @Min(0) @Max(100)
    private int discount;

    @CreatedDate
    @Setter @Getter
    private Date createdAt;

    @LastModifiedDate
    @Setter @Getter
    private Date updatedAt;

    public String getFullName() {
        return this.getName() + " " + this.getSurname();
    }
}
